logical/plugin/plugin.go

```diff
@@ -140,12 +140,10 @@ func newPluginClient(ctx context.Context, sys pluginutil.RunnerUtil, pluginRunne
 	}
 
 	// Wrap the backend in a tracing middleware
-	if logger.IsTrace() {
+	if logger.IsDebug() {
 		backend = &backendTracingMiddleware{
-			logger:    logger,
-			transport: transport,
-			typeStr:   pluginRunner.Name,
-			next:      backend,
+			logger: logger.With("plugin_type", pluginRunner.Name, "transport", transport),
```

Comment:
Since the backend will have type info already in the name I think plugin_type is redundant here.

Reply:
Currently the type info will just be "plugin"

Reply:
At creation time of the plugin we pass in this logger, right? So at plugin creation time can't we set an appropriate name?